tests/cases/conformance/types/thisType/thisTypeErrors2.ts

```diff
@@ -0,0 +1,12 @@
+class Base {
+    constructor(a: this) {
+    }
+}
+class Generic<T> {
+}
+class Derived {
+    n: number;
+    constructor(public host: Generic<this>) {

```

Comment:
Maybe use `this` in a type annotation of a local variable to test that `this` can be referenced in the body of a constructor.